### PR TITLE
MSC4156: Migrate server_name to via

### PR DESCRIPTION
**Pull request by Johennes**
Spec for [MSC4156](https://github.com/matrix-org/matrix-spec-proposals/pull/4156).

Relates to: https://github.com/matrix-org/matrix-spec/issues/884

### Pull Request Checklist

* [x] Pull request includes a [changelog file](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#adding-to-the-changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#sign-off)
* [ ] Pull request is classified as ['other changes'](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#other-changes)

Preview: https://pr1933--matrix-spec-previews.netlify.app

### CODE REVIEW

File: changelogs/client_server/newsfragments/1933.deprecation [hunk 1]
```diff
@@ -0,0 +1 @@
+Deprecate the `server_name` query parameter on `/_matrix/client/v3/join/{roomIdOrAlias}` and `/_matrix/client/v3/knock/{roomIdOrAlias}` as per MSC4156.
```
Comment (richvdh) [inline changelogs/client_server/newsfragments/1933.deprecation:1]:
```suggestion
Deprecate the `server_name` query parameter on `/_matrix/client/v3/join/{roomIdOrAlias}` and `/_matrix/client/v3/knock/{roomIdOrAlias}` as per [MSC4156](https://github.com/matrix-org/matrix-spec-proposals/pull/4156).
```

File: changelogs/client_server/newsfragments/1933.feature [hunk 2]
```diff
@@ -0,0 +1 @@
+Add `via` query parameter on `/_matrix/client/v3/join/{roomIdOrAlias}` and `/_matrix/client/v3/knock/{roomIdOrAlias}` as per MSC4156.
```
Comment (richvdh) [inline changelogs/client_server/newsfragments/1933.feature:1]:
```suggestion
Add `via` query parameter on `/_matrix/client/v3/join/{roomIdOrAlias}` and `/_matrix/client/v3/knock/{roomIdOrAlias}` as per [MSC4156](https://github.com/matrix-org/matrix-spec-proposals/pull/4156).
```

**Review by richvdh — APPROVED** [review]
LGTM otherwise

File: changelogs/client_server/newsfragments/1933.feature (hunk 2 above)
Comment (Johennes) [reply changelogs/client_server/newsfragments/1933.feature:1]: 🤦 Thank you.

**Pull request merged**